### PR TITLE
Change our DateTime marshalling tests to be culture-independent.

### LINKED ISSUE #24089: Test failure: PInvoke Interop\PInvoke\DateTime\DateTimeTest\DateTimeTest.cmd [FAIL]
Building and running tests on commit 837ac49c1ac46729df88ddb4a5d73ade6cc55cfc on Windows 10 Ent 1809 en-US with custom system wide date time format settings results in PInvoke DateTimeTest failure:

```
        Interop\PInvoke\DateTime\DateTimeTest\DateTimeTest.cmd [FAIL]

        Return code:      1
        Raw output file:      E:\src\ms\dotnet\coreclr\bin\tests\Windows_NT.x64.Checked\Reports\Interop.PInvoke\DateTime\DateTimeTest\DateTimeTest.output.txt
        Raw output:
        BEGIN EXECUTION
         "E:\src\ms\dotnet\coreclr\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\corerun.exe" DateTimeTest.exe
        Test Failure: TestLibrary.AssertTestException: Assert.IsTrue: Marshal_In_stdcall : Returned false
           at TestLibrary.Assert.HandleFail(String assertionName, String message) in E:\src\ms\dotnet\coreclr\tests\src\Common\CoreCLRTestLibrary\Assertion.cs:line 725
           at DatetimeTest.Main(String[] args) in E:\src\ms\dotnet\coreclr\tests\src\Interop\PInvoke\DateTime\DateTimeTest.cs:line 111
        FAILURE! InDATE expected '07/04/2008' but received: 2008-07-04
        Expected: 100
        Actual: 101
        END EXECUTION - FAILED
        FAILED
        Test Harness Exitcode is : 1
        To run the test:
        > set CORE_ROOT=E:\src\ms\dotnet\coreclr\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root
        > E:\src\ms\dotnet\coreclr\bin\tests\Windows_NT.x64.Checked\Interop\PInvoke\DateTime\DateTimeTest\DateTimeTest.cmd
        Expected: True
        Actual:   False
        Stack Trace:
          E:\src\ms\dotnet\coreclr\bin\tests\Windows_NT.x64.Checked\TestWrappers\Interop.PInvoke\Interop.PInvoke.XUnitWrapper.cs(5349,0): at Interop_PInvoke._DateTime_DateTimeTest_DateTimeTest_._DateTime_DateTimeTest_DateTimeTest_cmd()
        Output:

          Return code:      1
          Raw output file:      E:\src\ms\dotnet\coreclr\bin\tests\Windows_NT.x64.Checked\Reports\Interop.PInvoke\DateTime\DateTimeTest\DateTimeTest.output.txt
          Raw output:
          BEGIN EXECUTION
           "E:\src\ms\dotnet\coreclr\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\corerun.exe" DateTimeTest.exe
          Test Failure: TestLibrary.AssertTestException: Assert.IsTrue: Marshal_In_stdcall : Returned false
             at TestLibrary.Assert.HandleFail(String assertionName, String message) in E:\src\ms\dotnet\coreclr\tests\src\Common\CoreCLRTestLibrary\Assertion.cs:line 725
             at DatetimeTest.Main(String[] args) in E:\src\ms\dotnet\coreclr\tests\src\Interop\PInvoke\DateTime\DateTimeTest.cs:line 111
          FAILURE! InDATE expected '07/04/2008' but received: 2008-07-04
          Expected: 100
          Actual: 101
          END EXECUTION - FAILED
          FAILED
          Test Harness Exitcode is : 1
          To run the test:
          > set CORE_ROOT=E:\src\ms\dotnet\coreclr\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root
          > E:\src\ms\dotnet\coreclr\bin\tests\Windows_NT.x64.Checked\Interop\PInvoke\DateTime\DateTimeTest\DateTimeTest.cmd
```

Custom date time format set on system: 

Regional date time format settings

`dotnet --info` output:

```
dotnet --info
Tools are already initialized.
Running: dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   2.1.503
 Commit:    4c506e0f35

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   E:\src\ms\dotnet\coreclr\.dotnet\sdk\2.1.503\

Host (useful for support):
  Version: 2.1.7
  Commit:  cca5d72d48

.NET Core SDKs installed:
  2.1.503 [E:\src\ms\dotnet\coreclr\.dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.7 [E:\src\ms\dotnet\coreclr\.dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.7 [E:\src\ms\dotnet\coreclr\.dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.7 [E:\src\ms\dotnet\coreclr\.dotnet\shared\Microsoft.NETCore.App]

```

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Rewrite our DateTime marshalling tests to not use Windows culture-sensitive APIs to stringify DATE's to verify that they are correct. Instead, do much simpler native calls that only depend on the managed value. Specifically target the tests to it each of the code-paths of the DateTime marshalers instead of adding other tests that just duplicate coverage in other PInvoke tests.

Fixes #24089

### CODE REVIEW

File: tests/src/Interop/PInvoke/DateTime/DateTimeTest.cs [hunk 1]
```diff
@@ -7,172 +7,46 @@
...
-
-            Stru_Seq_DateAsStructAsFld StDate1;
-            StDate1.dt = new DateTime(2008, 7, 4);
-            StDate1.iInt = 100;
-            StDate1.bstr = "Managed";
-            Assert.IsTrue(MarshalSeqStruct_InOut_cdecl(ref StDate1), "MarshalSeqStruct_InOut_cdecl : Native side check failed");
-            Assert.AreEqual(ExpectedRetdate, StDate1.dt, "MarshalSeqStruct_InOut_cdecl : Returned date is wrong");
+        try
+        {
+            DateTime currentDate = new DateTime(2019, 5, 2);
```
Comment (AaronRobinsonMSFT) [inline tests/src/Interop/PInvoke/DateTime/DateTimeTest.cs]: Instead of hardcoding this, it might be beneficial to instead use `DateTime.Now`. The benefit of this is that it is predictable _and_ we will properly test all cases through time (i.e. leap year, month nonsense, etc). Would any of those things even apply?

Comment (AaronRobinsonMSFT) [reply tests/src/Interop/PInvoke/DateTime/DateTimeTest.cs]: You could also start from `DateTime.Now` and in a loop go through 366 days to test all the scenarios for a year in the future. Unsure how long the test would take though.

Comment (jkoritzinsky) [reply tests/src/Interop/PInvoke/DateTime/DateTimeTest.cs]: I don't think that date-time nonsense would really apply here. Under the hood the conversion is just converting to/from the number of "ticks". The actual date-time nonsense should be tested in our tests for the DateTime structure.
